### PR TITLE
Replace anonymous function with class method

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Remove closure in discovery class. 

## Testing Instructions

---

See #1117

### CODE REVIEW

File: includes/Discovery.php [hunk 1]
```diff
@@ -70,6 +62,17 @@ static function () {
 		add_action( 'web_stories_story_head', 'wp_oembed_add_discovery_links' );
 	}
 
+	/**
+	 * Theme support for title-tag is implied for stories. See _wp_render_title_tag().
+	 *
+	 * @return void
+	 */
+	public function print_title_metadata() {
```
Comment (swissspidy) [inline includes/Discovery.php]: Can you rename this to `print_metadata`? #2336

I don't want a method for every single field. `print_metadata` for title & description sounds good enough.

```suggestion
	public function print_metadata() {
```

File: includes/Discovery.php [hunk 2]
```diff
@@ -41,15 +41,7 @@ class Discovery {
...
-		add_action(
-			'web_stories_story_head',
-			static function () {
-				// Theme support for title-tag is implied for stories. See _wp_render_title_tag().
-				echo '<title>' . esc_html( wp_get_document_title() ) . '</title>' . "\n";
-			},
-			1
-		);
-
+		add_action( 'web_stories_story_head', [ $this, 'print_title_metadata' ], 1 );
```
Comment (swissspidy) [inline includes/Discovery.php]:
```suggestion
		add_action( 'web_stories_story_head', [ $this, 'print_metadata' ], 1 );
```

File: includes/Discovery.php [hunk 3]
```diff
@@ -70,6 +62,17 @@ static function () {
 		add_action( 'web_stories_story_head', 'wp_oembed_add_discovery_links' );
 	}
 
+	/**
+	 * Theme support for title-tag is implied for stories. See _wp_render_title_tag().
```
Comment (swissspidy) [inline includes/Discovery.php]:
```suggestion
	 * Prints general metadata on the single story template. 
	 *
	 * Theme support for title tag is implied for stories.
	 *
	 * @see _wp_render_title_tag().
```

**Review by swissspidy — COMMENTED** [review]
Could have fit into #2336, but alas. Let's settle on the method name at least.

**Comment by spacedmonkey** [comment]
> Could have fit into #2336, but alas. Let's settle on the method name at least.

I thought of it after, sorry.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**